raiden/network/transport/matrix/transport.py

```diff
@@ -570,9 +570,6 @@ def _run(self) -> None:  # type: ignore
             self._stop_event.set()
             gevent.killall(self.greenlets)  # kill children
             raise  # re-raise to keep killed status
-        except Exception:
-            self.stop()  # ensure cleanup and wait on subtasks
```

Comment:
Previously, this would actually deadlock, if there were still items in the `broadcast_queue` -- the `broadcast_worker` would have been killed already, but `self.stop()` would call `broadcast_queue.join()`.